Break out users by version numbers for update orphaning dashboard

sapohl opened the pull request:
needs-review: @vitillo

Tracked in https://bugzilla.mozilla.org/show_bug.cgi?id=1284899

The dashboard has already been updated to accommodate the additional data, and to gracefully handle our old .json files that did not include data about version distribution.

sapohl committed:
Break out users by version numbers for update orphaning dashboard

Tracked in https://bugzilla.mozilla.org/show_bug.cgi?id=1284899

sapohl committed:
Change version distribution analysis to focus on orphaned users with updates enabled

Previously, we were mistakenly analyzing the distribution across
Firefox versions of all orphaned users. This should have been focusing
on orphaned users with updates enabled only.

I also used this opportunity to improve the message in case we assert
because we don’t have a longitudinal dataset available that’s recent
enough for our analysis.

sapohl committed:
Break out orphaned users by usage hours

Break out orphaned users who haven't run a particular orphaned version
of Firefox for at least 2 hours.

sapohl committed:
Merge pull request #1 from sapohl/two-hour-usage-feature

Break out orphaned users by usage hours

sapohl committed:
Revert "Break out orphaned users by usage hours"

sapohl committed:
Merge pull request #2 from sapohl/revert-1-two-hour-usage-feature

Revert "Break out orphaned users by usage hours"

vitillo commented on reports/update-orphaning/Update orphaning analysis using longitudinal dataset.ipynb:
```diff
@@ -263,7 +266,52 @@
+    "def version_mapper(d):\n",
+    "    status, ping = d\n",
+    "    if ping.version[0] is None:\n",
+    "        return (\"-1\", ping)\n",
```
I think `None` should work as well as key and make more sense than an arbitrary string.

sapohl committed:
Remove unnecessary None-check

This check is actually unnecessary, since |status_mapper| guarantees that ping.version[0] is non-None.

sapohl commented:
Removed unnecessary None-check

vitillo commented:
lgtm
